Wrappers/Python/cil/optimisation/algorithms/SIRT.py

```diff
@@ -133,11 +134,18 @@ def fix_weights(self):
         
         # fix for possible +inf, -inf, nan values 
         for arr in [self.M, self.D]:  
- 
-            tmp = arr.as_array()
-            arr_replace = numpy.isfinite(tmp)
-            tmp[~arr_replace] = 1.0     
-            arr.fill(tmp)                     
+            self._remove_nan_or_inf(arr, replace_with=1.0)
+
+    def _remove_nan_or_inf(self, datacontainer, replace_with=1.0):
```

Comment:
Add comment to explain this is to make it work for block data container.

Reply:
Also state what the fix consists of.

Reply:
if a `BlockDataContainer` is passed then the method makes a recursive call.